### ISSUE TITLE
Simpler installation

### ISSUE THREAD

**Issue by amir-zeldes** [gucorpling/gitdox#70]
User reports suggest installing on a new server is complicated, esp. due to npm, EtherCalc installation, dealing with SELinux.

**Comment by amir-zeldes** [comment]
Possible solution using docker?

**Comment by amir-zeldes** [comment]
Yes, github3 is a dependency if we want to be able to use the commit functionality, but EtherCalc is the really tricky part for admins trying to install this.

**Comment by amir-zeldes** [comment]
For docker support we need to figure out how to persist user information and config

**Comment by lgessler** [comment]
It's easier than we thought!! We just didn't understand how Docker works. After you create an instance of a Docker image, file changes are automatically persisted. And to edit files manually, you can use Docker to get a bash session inside the container. 

This is our Dockerfile:
```Dockerfile
FROM ubuntu:18.04
EXPOSE 80

# install deps
RUN apt update
RUN apt -y install apache2 python python-pip redis-server gzip git curl libssl-dev pkg-config build-essential npm libxml2-utils git

# pull gitdox and give permissions
RUN rm -rf /var/www/html
RUN git clone https://github.com/gucorpling/gitdox.git /var/www/html
RUN chown -R www-data:www-data /var/www/html
RUN chmod +x /var/www/html/*.py
RUN chmod +x /var/www/html/modules/*.py

# keep these in sync with requirements.txt
RUN pip install lxml requests github3.py==0.9.3 passlib

# start ethercalc
RUN npm install -g ethercalc
RUN ethercalc &

# enable cgi
RUN a2enmod cgi
RUN echo "                       \n \
<Directory /var/www/html>        \n \
   Options +ExecCGI              \n \
   AddHandler cgi-script .py     \n \
   DirectoryIndex index.py       \n \
</Directory>                     \n \
" >> /etc/apache2/apache2.conf
RUN service apache2 restart

CMD /usr/sbin/apache2ctl -D FOREGROUND
```

You can interact with it like this:

```bash
# execute the Dockerfile in the current directory (.) to set up the image 
# which will then be called gitdox and stored in your machine's image registry 
docker build -t gitdox .

# create and start a new instance of the image called "gitdox" (--name gitdox),
# detach it from your shell (-d), allow interactive sessions (-it), map container's port 80
# to your machine's port 5000 (-p 5000:50) and call the instantiation of the gitdox image
# gitdox-instance
docker run -d -it --name gitdox -p 80:80 gitdox-instance

# start and stop using these commands--the instance persists no matter what unless you
# explicitly docker rm it
docker stop gitdox-instance
docker start gitdox-instance

# use to start a bash session inside the container
docker exec -it gitdox-instance bash
```

**Comment by amir-zeldes** [comment]
This makes total sense, come to think of it I've actually done something like this before to retrieve apache logs from a container after a crash (you can literally exec tail into the container and get out the state from the file system)

**Comment by amir-zeldes** [comment]
Implemented in 80d615552866831a29256038443e7974ebc9d4bf
